src/prefect/tasks/aws/lambda_function.py

```diff
@@ -0,0 +1,407 @@
+            )
+            return response
+
+        response = lambda_client.delete_function(FunctionName=self.function_name)
+        return response
+
+
+class LambdaInvoke(Task):
+    """
+    Task to invoke a Lambda function.
+
+    Args:
+        - function_name (str): the name of the Lambda funciton to invoke
+        - invocation_type (str, optional): the invocation type of Lambda
+            function, default is RequestResponse other options include
+            Event and DryRun
+        - log_type (str, optional): set to 'Tail' to include the execution
+            log in the response
+        - client_context (dict, optional): data to pass to the function in the
+            context object, dict object will be transformed into base64 encoded
+            json automatically
+        - payload (bytes or seekable file-like object): the JSON provided to
+            Lambda function as input
+        - qualifier (str, optional): specify a version or alias to invoke a
+            published version of the function, defaults to $LATEST
+        - aws_credentials_secret (str, optional): the name of the Prefect Secret
+            which stores your AWS credentials; this Secret must be a JSON string
+            with two keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY`
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        function_name: str,
+        invocation_type: str = "RequestResponse",
+        log_type: str = "None",
+        client_context: dict = None,
+        payload: str = json.dumps(None),
+        qualifier: str = "$LATEST",
+        aws_credentials_secret: str = "AWS_CREDENTIALS",
+        **kwargs
+    ):
+        self.function_name = function_name
+        self.invocation_type = invocation_type
+        self.log_type = log_type
+
+        ## encode input dictionary as base64 json
+        self.client_context = self._encode_lambda_context(**(client_context or {}))
+
+        self.payload = payload
+        self.qualifier = qualifier
+        self.aws_credentials_secret = aws_credentials_secret
+        super().__init__(**kwargs)
+
+    def _encode_lambda_context(self, custom=None, env=None, client=None):
+
+        """
+        Utility function for encoding Lambda context
+
+        Args:
+            - custom (dict, optional): key-value  pairs to pass to custom context
+            - env (dict, optional): key-value pairs to pass to environment context
+            - client (dict, optional): key-value pairs to pass to client context
+
+        Returns:
+            - json: base64 encoded json object
+        """
+
+        client_context = dict(custom=custom, env=env, client=client)
+        json_context = json.dumps(client_context).encode("utf-8")
+        return b64encode(json_context).decode("utf-8")
+
+    @defaults_from_attrs("aws_credentials_secret", "payload")
```

Comment:
If you add "function_name" to this list, then you could determine the function at runtime based on the output of an upstream task. For example, you could return a list of function names and payloads from task A and then map over them from this task without having to specify the function name in advance - which could be pretty cool.

Reply:
Good idea, I added this parameter in the latest batch of commits